gremlin-go/driver/bytecode.go

```diff
@@ -30,7 +30,7 @@ type bytecode struct {
 	bindings           map[string]interface{}
 }
 
-func newBytecode(bc *bytecode) *bytecode {
+func NewBytecode(bc *bytecode) *bytecode {
```

Comment:
I worry about this change. It looks like you're exporting it now to use with NewGraphTraversal?

Is there a better way of doing this? Returning unexported types is not recommended and can cause issues. One of which is that documentation for it cannot be generated, as well as making usage of the method esoteric.

Reply:
Yes, it is so we can extend the anonymous traversal functions to add steps. The start of each anonymous traversal needs a new bytecode object, so I'm not sure if there is a way to use it without exporting this...

Reply:
In any case, I'll see if there are other way to wrap this without exporting it. 
Do you have any comments regarding the other exports or is this one the only concern?